### PR TITLE
Add dropout to BertClassifier

### PR DESCRIPTION
**Pull request by chenmoneygithub**

### CODE REVIEW

File: keras_nlp/models/bert/bert_tasks.py [hunk 1]
```diff
@@ -262,6 +264,7 @@ def from_preset(
         # Check if preset is backbone-only model
         if preset in Bert.presets:
             backbone = Bert.from_preset(preset, load_weights)
+            kwargs["dropout"] = Bert.presets[preset]["config"]["dropout"]
```
Comment (jbischof) [inline keras_nlp/models/bert/bert_tasks.py:267]: Why do we need this line? This function lets the user set a separate `dropout` for the classifier and the you override it silently.

File: keras_nlp/models/bert/bert_tasks.py [hunk 2]
```diff
@@ -101,11 +101,13 @@ def __init__(
         self,
         backbone,
         num_classes=2,
+        dropout=0.1,
```
Comment (mattdangerw) [inline keras_nlp/models/bert/bert_tasks.py:104]: Can we just not expose this for now and read the property of the backbone?

```
pooled = backbone(inputs)["pooled_output"]
pooled = keras.layers.Dropout(backbone.dropout)(pooled)
```

File: keras_nlp/models/bert/bert_tasks.py (hunk 1 above)
Comment (chenmoneygithub) [reply keras_nlp/models/bert/bert_tasks.py:267]: In this if branch it reads the BERT presets instead of the classifier presets, so it does not have the `dropout` value. configured in classifier presets.

File: keras_nlp/models/bert/bert_tasks.py (hunk 2 above)
Comment (chenmoneygithub) [reply keras_nlp/models/bert/bert_tasks.py:104]: I think we should give users freedom to adjust the dropout value?

Comment (mattdangerw) [reply keras_nlp/models/bert/bert_tasks.py:104]: If we want it configurable, you could still have it default to the backbone value (even for our presets). Something like...

```python
def __init__(..., dropout=None):
    ...
    pooled = backbone(inputs)["pooled_output"]
    # Default to the same dropout value as our backbone
    dropout = dropout or backbone.dropout
    pooled = keras.layers.Dropout(dropout)(pooled)
```

This was you would not need to double up the parameter in the preset config. But if someone for some reason wanted a different value, you would have the flexibility to do so.

Comment (jbischof) [reply keras_nlp/models/bert/bert_tasks.py:104]: Would anyone confuse `dropout=None` with `dropout=0.0`?

Comment (mattdangerw) [reply keras_nlp/models/bert/bert_tasks.py:104]: Perhaps!

Also just remembered we have this already on all our other heads. So maybe scratch my whole suggestion at least just for consistency.

Comment (chenmoneygithub) [reply keras_nlp/models/bert/bert_tasks.py:104]: I don't think defaulting the dropout value to backbone is necessary, actually I think it could create confusion on users because this default is hidden logic. 

See TFM's implementation: https://www.tensorflow.org/api_docs/python/tfm/nlp/models/BertClassifier

Having an explicit default is clear.

Comment (mattdangerw) [reply keras_nlp/models/bert/bert_tasks.py:104]: Fine with me! But then we should remove the line Jonathan is suggesting above.

File: keras_nlp/models/bert/bert_presets.py [hunk 3]
```diff
@@ -238,6 +238,7 @@
                 },
             },
             "num_classes": 2,
+            "dropout": 0.1,
```
Comment (mattdangerw) [inline keras_nlp/models/bert/bert_presets.py:241]: We should probably retrain with this change right (unless it makes no difference?). Should we open a follow up?
